buildtest/modules/spider.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import subprocess
+
+from buildtest.tools.defaults import BUILDTEST_SPIDER_FILE, BUILDTEST_CONFIG_FILE
+from buildtest.tools.config import load_configuration
+
+
+class Spider:
+    """Class declaration of Spider class"""
+
+    def __init__(self, tree=None):
+        """Initialize method for Spider class.
+
+        :param tree: User can specify a module tree to query from spider.
+        :type tree: str
+        """
+
+        #  if user specifies a tree, then run spider command otherwise read from configuration file.
+        if tree:
+            self.tree = tree
+            spider_cmd = f"{os.getenv('LMOD_DIR')}/spider -o spider-json {self.tree}"
+
+            out = subprocess.check_output(spider_cmd, shell=True).decode("utf-8")
+            self.spider_content = json.loads(out)
+
+        else:
+            with open(BUILDTEST_SPIDER_FILE, "r") as fd:
+                self.spider_content = json.load(fd)
+
+            content = load_configuration(BUILDTEST_CONFIG_FILE)
+            self.tree = content["BUILDTEST_MODULEPATH"]
+
+    def get_trees(self):
+        """" Return module trees used in spider command
+
+        :return: return module trees used for querying from spider
+        :rtype: str
+        """
+        return self.tree
+
+    def get_unique_software(self):
+        """Return all keys from spider. This is the unique software names.
+
+        :return: return sorted list of all spider keys.
+        :rtype: list
+        """
+
+        return sorted(list(self.spider_content.keys()))
```

Comment:
You don't need the list here, the sorted keys already is a list. Here is an example:

```
test = {1:1, 2:2, 3:3}
sorted(test)                                                                                                            
[1, 2, 3]

type(sorted(test))                                                                                                      
list
```